app/src/main/java/de/tum/in/tumcampusapp/component/tumui/roomfinder/RoomFinderDetailsActivity.kt

```diff
@@ -0,0 +1,300 @@
+package de.tum.`in`.tumcampusapp.component.tumui.roomfinder
+
+import android.content.ActivityNotFoundException
+import android.content.DialogInterface
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.Fragment
+import de.tum.`in`.tumcampusapp.R
+import de.tum.`in`.tumcampusapp.api.app.ApiHelper
+import de.tum.`in`.tumcampusapp.api.app.TUMCabeClient
+import de.tum.`in`.tumcampusapp.component.other.generic.ImageViewTouchFragment
+import de.tum.`in`.tumcampusapp.component.other.generic.activity.ActivityForLoadingInBackground
+import de.tum.`in`.tumcampusapp.component.other.locations.LocationManager
+import de.tum.`in`.tumcampusapp.component.other.locations.model.Geo
+import de.tum.`in`.tumcampusapp.component.tumui.roomfinder.model.RoomFinderCoordinate
+import de.tum.`in`.tumcampusapp.component.tumui.roomfinder.model.RoomFinderMap
+import de.tum.`in`.tumcampusapp.component.tumui.roomfinder.model.RoomFinderRoom
+import de.tum.`in`.tumcampusapp.utils.Const
+import de.tum.`in`.tumcampusapp.utils.NetUtils
+import de.tum.`in`.tumcampusapp.utils.Utils
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+/**
+ * Displays the map regarding the searched room.
+ */
+class RoomFinderDetailsActivity : ActivityForLoadingInBackground<Void, String>(R.layout.activity_roomfinderdetails), DialogInterface.OnClickListener {
+
+    lateinit var mImageFragment: ImageViewTouchFragment
+
+    private var mapsLoaded: Boolean = false
+
+    lateinit var room: RoomFinderRoom
+    private var mapId: String = ""
+    private var mapsList: List<RoomFinderMap> = ArrayList()
+    private val infoLoaded: Boolean = false
+
+    private var fragment: Fragment? = null
+
+    private var mRoomFinderCoordinateCall: Call<RoomFinderCoordinate>? = null
+    private var mRoomFinderMapsCall: Call<List<RoomFinderMap>>? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        mImageFragment = ImageViewTouchFragment.newInstance()
+        supportFragmentManager.beginTransaction()
+                .add(R.id.fragment_container, mImageFragment)
+                .commit()
+
+        room = intent.getSerializableExtra(EXTRA_ROOM_INFO) as RoomFinderRoom
```

Comment:
You can move this to a lazy property. 

```kotlin
private val room: RoomFinderRoom by lazy {
    intent.getSerializableExtra(EXTRA_ROOM_INFO) as RoomFinderRoom
}
```

Then, you could also immediately initialize the `Call` objects to avoid dealing with their nullability.